### ISSUE TITLE
Ouput difference between with_query and update_query

### ISSUE THREAD

**Issue by dimaqq** [aio-libs/yarl#378]
```
In [2]: from yarl import URL

In [6]: str(URL("/").with_query(a="foo%20bar"))
Out[6]: '/?a=foo%20bar'

In [7]: str(URL("/").update_query(a="foo%20bar"))
Out[7]: '/?a=foo+bar'
```

This is kindof unexpected...
`yarl==1.3.0`

**Comment by dimaqq** [comment]
I wonder if this was already fixed by #339 🤔

**Comment by asvetlov** [comment]
Would you check?

**Comment by dimaqq** [comment]
Same with `yarl==1.4.0a11`:

```
(xx) … ~> pip install yarl
...
Successfully installed idna-2.8 multidict-4.5.2 yarl-1.3.0
(xx) … ~> python -c 'import yarl; print(yarl.URL("/").with_query(a="foo%20bar"), yarl.URL("/").update_query(a="foo%20bar"))'
/?a=foo%20bar /?a=foo+bar
(xx) … ~> pip install yarl==1.4.0a11
...
Successfully installed yarl-1.4.0a11
(xx) … ~> python -c 'import yarl; print(yarl.URL("/").with_query(a="foo%20bar"), yarl.URL("/").update_query(a="foo%20bar"))'
/?a=foo%20bar /?a=foo+bar
```

**Comment by dimaqq** [comment]
I'll try master branch, give me a sec, on macOS and the build setup is a bit odd...

**Comment by dimaqq** [comment]
```
(xx) … ~> git clone git@github.com:aio-libs/yarl.git
...
(xx) … ~> pip uninstall yarl
...
(xx) … ~> cd yarl/
(xx) … ~/yarl> make cythonize
... (ok)
(xx) … ~/yarl> pip install -e .
... (ok)
(xx) … ~/yarl> python -c 'import yarl; print(yarl.URL("/").with_query(a="foo%20bar"), yarl.URL("/").update_query(a="foo%20bar"))'
/?a=foo%20bar /?a=foo+bar
```

**Comment by asvetlov** [comment]
I agree this is confusing.

**Comment by asvetlov** [comment]
Fixed by #472
